### PR TITLE
Chartjs upgrade v4.2.1

### PR DESCRIPTION
**Pull request by jamiecarter7**
**Story card:** [sc-9888](https://app.shortcut.com/simpledotorg/story/9888/upgrade-chartjs-to-latest-version-v4)

## Because

While working in this area bring ChartJS to the latest version.

## This addresses

Bring ChartJS to the latest version for future proofing

## Test instructions

Testing data against master branch to ensure that figures are identical

### CODE REVIEW

File: app/assets/javascripts/common/my_facilities.js [hunk 1]
```diff
@@ -54,7 +54,6 @@ function createBaseTrendChartConfig() {
           radius: 0,
         },
       },
-      responsive: true,
```
Comment (jamiecarter7) [inline app/assets/javascripts/common/my_facilities.js:57]: Responsive default values = true 

This is not required
https://www.chartjs.org/docs/3.9.1/configuration/responsive.html

File: app/assets/javascripts/common/reports.js [hunk 2]
```diff
@@ -1094,10 +1093,8 @@ function baseLineGraphConfig() {
       scales: {
         x: {
           stacked: false,
-          display: true,
           grid: {
             display: false,
-            drawBorder: true,
```
Comment (jamiecarter7) [inline app/assets/javascripts/common/reports.js:1100]: ```scales[id].grid.drawBorder has been renamed to scales[id].border.display```
https://www.chartjs.org/docs/latest/migration/v4-migration.html

`options.scales[scaleId].border`
this display value is default = true
removed as not required
https://www.chartjs.org/docs/latest/axes/styling.html#border-configuration

File: app/assets/javascripts/common/reports.js [hunk 3]
```diff
@@ -1094,10 +1093,8 @@ function baseLineGraphConfig() {
       scales: {
         x: {
           stacked: false,
-          display: true,
```
Comment (jamiecarter7) [inline app/assets/javascripts/common/reports.js:1097]: default display = true
removed as not required
https://www.chartjs.org/docs/latest/axes/#common-options-to-all-axes

File: app/assets/javascripts/common/reports.js [hunk 4]
```diff
@@ -1106,16 +1103,15 @@ function baseLineGraphConfig() {
...
           },
           beginAtZero: true,
           min: 0,
         },
         y: {
           stacked: false,
-          display: true,
-          grid: {
-            display: true,
-            drawBorder: false,
```
Comment (jamiecarter7) [inline app/assets/javascripts/common/reports.js:1118]: default grid - display = true 
removed as not required 
https://www.chartjs.org/docs/latest/axes/styling.html#grid-line-configuration

File: app/assets/javascripts/common/reports.js [hunk 5]
```diff
@@ -1106,16 +1103,15 @@ function baseLineGraphConfig() {
               family: "Roboto",
             },
             padding: 6,
+            showLabelBackdrop: true,
```
Comment (jamiecarter7) [inline app/assets/javascripts/common/reports.js:1106]: upgrading to v4.2.1 moves the dates on the x axis to the left slightly.
Seems like they line up better when dashes are shown on the x-axis.
Adding this re-aligns the labels back to where they were positioned in v3.9.1
Screenshot 2023-03-21 at 11 20 55
Screenshot 2023-03-21 at 11 20 13

I've submitted an error report about this - will keep an eye on this.
https://github.com/chartjs/Chart.js/issues/11197

**Review by harimohanraj89 — APPROVED** [review]
Looks like a very smooth upgrade, nice work!

**Comment by harimohanraj89** [comment]
NVM you're way ahead of me

**Pull request merged**